src/confluent_kafka/src/AdminTypes.c

```diff
@@ -73,19 +73,33 @@ static int NewTopic_init (PyObject *self0, PyObject *args,
                                "replica_assignment",
                                "config",
                                NULL };
-
+                               
+        self->num_partitions = -1;
         self->replication_factor = -1;
         self->replica_assignment = NULL;
         self->config = NULL;
 
-        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "si|iOO", kws,
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|iiOO", kws,
                                          &topic, &self->num_partitions,
                                          &self->replication_factor,
                                          &self->replica_assignment,
                                          &self->config))
                 return -1;
 
 
+        if(self->replica_assignment){
```

Comment:
Let's not add these check. Keep the existing checks only. These checks should be added in librdkafka or broker side as these are applicable only for v4 of CreateAPI but we are supporting all the versions.